hipcheck/src/policy_exprs/env.rs

```diff
@@ -395,32 +433,58 @@ fn neq(env: &Env, args: &[Expr]) -> Result<Expr> {
 		(Int(arg_1), Int(arg_2)) => Ok(Bool(arg_1 != arg_2)),
 		(Float(arg_1), Float(arg_2)) => Ok(Bool(arg_1 != arg_2)),
 		(Bool(arg_1), Bool(arg_2)) => Ok(Bool(arg_1 != arg_2)),
+		(DateTime(arg_1), DateTime(arg_2)) => Ok(Bool(arg_1 != arg_2)),
+		(Span(arg_1), Span(arg_2)) => Ok(Bool(
+			arg_1
+				.compare(arg_2)
+				.map_err(|err| Error::Datetime(err.to_string()))?
+				.is_ne(),
+		)),
 		_ => unreachable!(),
 	};
 
 	binary_primitive_op(name, env, args, op)
 }
 
+// Adds numbers or adds a span of time to a datetime (the latter use case is *not* commutative)
+// Datetime addition will error for spans with units greater than days (which the parser should prevent)
 fn add(env: &Env, args: &[Expr]) -> Result<Expr> {
 	let name = "add";
 
 	let op = |arg_1, arg_2| match (arg_1, arg_2) {
 		(Int(arg_1), Int(arg_2)) => Ok(Int(arg_1 + arg_2)),
 		(Float(arg_1), Float(arg_2)) => Ok(Float(arg_1 + arg_2)),
 		(Bool(_), Bool(_)) => Err(Error::BadType(name)),
+		(DateTime(arg_1), Span(arg_2)) => Ok(DateTime(
+			arg_1
+				.checked_add(arg_2)
+				.map_err(|err| Error::Datetime(err.to_string()))?,
+		)),
+		(DateTime(arg_1), DateTime(arg_2)) => Err(Error::BadType(name)),
+		(Span(arg_1), Span(arg_2)) => Err(Error::BadType(name)),
```

Comment:
Semantically it seems fine to add two spans (e.g. 2 days plus 4 weeks = 4w2d). Does the `Span` API not allow for that?

Reply:
Oh we can do that. But you didn't list it in one of the features we wanted, so I didn't include it. I can add that in.